src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -335,7 +342,7 @@ public String getRepositoryUri(@NonNull BitbucketRepositoryProtocol protocol,
         return pullRequests;
     }
 
-    private void setupPullRequest(BitbucketServerPullRequest pullRequest, BitbucketServerEndpoint endpoint) throws IOException {
+    private void setupPullRequest(BitbucketServerPullRequest pullRequest, BitbucketServerEndpoint endpoint) throws IOException, InterruptedException {
```

Comment:
Instead of catching `InterruptedException` and just propagating it as a flag with `Thread.interrupt()` like in #483, I chose to just propagate the `InterruptedException` directly since the `BitbucketApi` interface that this class implements already has `throws InterruptedException` in its signature.